src/core/agent/services/keriaNotificationService.ts

```diff
@@ -358,35 +357,30 @@ class KeriaNotificationService extends AgentService {
     }
 
     if (telStatus === Ilks.rev) {
-      if (!existingCredential) {
-        const notifications = await this.notificationStorage.findAllByQuery({
-          route: NotificationRoute.ExnIpexGrant,
-          read: false,
-        })
-          
-        if (notifications.length) {
-          const notificationRecord = notifications[0];
-          await this.deleteNotificationRecordById(
-            notificationRecord.id,
-            notificationRecord.a.r as NotificationRoute
-          );
+      const notifications = await this.notificationStorage.findAllByQuery({
+        route: NotificationRoute.ExnIpexGrant,
```

Comment:
@iFergal I’ve tried logging the data of exchange and notification in storage to find a value that can accurately identify the notification to be deleted, but I haven’t found it yet. The only common identifier available is the issuer. Do you have any suggestions on how to determine this?

Reply:
@Sotatek-BaoHoanga They will be completely independent grant messages. A notification record alone won't have enough information on this.

Perhaps the cleanest solution instead of iterating over all notifications is to add an optional `credentialId` tag to a notification, and for grant messages we can populate this field?